ipaserver/plugins/groupfas.py

```diff
@@ -173,16 +176,13 @@ def group_find_fas_postcb(self, ldap, entries, truncated, *args, **options):
 def group_mod_fas_precb(self, ldap, dn, entry, *keys, **options):
     """Add fasgroup object class and related attributes.
     """
-    fas_params = [
-        option for option in options
-        if option.startswith("fas") and option != "fasgroup"
-        # fasgroup is a flag, it's always present: filter it out.
-    ]
-    if options.get("fasgroup", False) or fas_params:
-        entry_oc = ldap.get_entry(dn, ["objectclass"])["objectclass"]
-        if not self.obj.has_objectclass(entry_oc, "fasgroup"):
+    if any(option.startswith("fas") for option in options):
+        # add fasgroup object class
+        if "objectclass" not in entry:
+            entry_oc = ldap.get_entry(dn, ["objectclass"])
+            entry["objectclass"] = entry_oc["objectclass"]
+        if not self.obj.has_objectclass(entry["objectclass"], "fasgroup"):
             entry_oc.append("fasgroup")
```

Comment:
I understand it'll work because `entry["objectclass"]` will be pointing at the same list reference, but it would maybe be clearer if `entry["objectclass"]` was modifed rather than the other variable.